Cargo.toml

```diff
@@ -14,9 +14,10 @@ keywords = ["cnab"]
 maintenance = { status = "experimental" }
 
 [dependencies]
-semver = { version = "0.9", features = ["serde"] }
-serde = { version = "1.0", features = ["derive"] }
-serde_json = "1.0"
+canonical_json = { git = "https://github.com/engineerd/canonical_json" }
+serde = "0.8"
+serde_derive = "0.8"
+serde_json = "0.8"
```

Comment:
The main caveat of this PR is that the canonical JSON implementation is currently only compatible with version 0.8 of Serde, meaning that for this PR, I had to:

- downgrade Serde to v0.8 from v1.0
- add again `serde_derive` again as a separate import
- temporarily remove the SemVer library, as I think it was incompatible with v0.8 of Serde (I have to check about this again, not sure)

That being said, this canonical JSON library (I forked an unmaintained library, and updated it for Rust 1.35 nightly - while the plan is to upgrade it to use Serde v1.0, I'm not really sure when that will be possible) might be the best implementation we have for a while, and if we accept this PR, we might have to stick with this version of Serde.

At the same time, canonical JSON is part of the core specification, so any library that wants to implement the core specification needs to have support for canonical JSON.

@jlegrone, @rylev, @technosophos, @glyn - what do you think?

Reply:
For reference, I've built a small binary to test the canonical JSON implementation (see https://github.com/engineerd/canonical_json), and I've been comparing it with the output from the canonical JSON used in the Go implementation (see https://github.com/technosophos/canonjson).

Reply:
> @jlegrone, @rylev, @technosophos, @glyn - what do you think?

I agree with downgrading serde as a _temporary_ expedient, but I wouldn't like us to get stuck there. I think we should look into upgrading an existing canonical json library to be serde 1.0 compatible in parallel. Or, if needed, write our own. How hard can it be? ;-)

Reply:
I’m more inclined not to get this merged in its current state,  at least not before upgrading the canonical JSON library.